tests: net: lib: lwm2m: content_oma_tlv: move to new ztest API

hunterhu3000 opened the pull request:
Move net content_oma_tlv tests to use new ztest API
Split all the test cases into 3 test suites, each test suite has
its own before function.

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

enjiamai reviewed (dismissed):
I am not sure if before/after can be applied here. LGTM.

hunterhu3000 commented:
> I am not sure if before/after can be applied here. LGTM.

if add before/after, there will be 2 test suites

enjiamai commented:
> if add before/after, there will be 2 test suites

Hi @hunterhu3000 , I think it ok to have 2 test suites, Ex. one is with  net_content_oma_tlv, and one is net_content_oma_tlv_nomen suffix, then the test_prepare and test_prepare_nomem could work as the "before" function. Ho do you think?

aaronemassey commented on tests/net/lib/lwm2m/content_oma_tlv/src/main.c:
```diff
@@ -1083,84 +1143,15 @@ static void test_get_objlnk(void)
-		ztest_unit_test_setup_teardown(
-			test_put_s8, test_prepare, unit_test_noop),
-		ztest_unit_test_setup_teardown(
-			test_put_s8_nomem, test_prepare_nomem, unit_test_noop),
```
I agree with @enjiamai. These should be separated as two suites with two different before. They'll both run since they're both in the same binary.

hunterhu3000 replied:
OK, updated. Thanks to review

hunterhu3000 committed:
tests: net: lib: lwm2m: content_oma_tlv: move to new ztest API

Move net content_oma_tlv tests to use new ztest API
Split all the test cases into 3 test suites, each test suite has
its own before function.

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>
